llvm/lib/Target/X86/X86IndirectThunks.cpp

```diff
@@ -63,77 +135,34 @@ class X86RetpolineThunks : public MachineFunctionPass {
   }
 
 private:
-  MachineModuleInfo *MMI;
-  const TargetMachine *TM;
-  bool Is64Bit;
-  const X86Subtarget *STI;
-  const X86InstrInfo *TII;
-
-  bool InsertedThunks;
-
-  void createThunkFunction(Module &M, StringRef Name);
-  void insertRegReturnAddrClobber(MachineBasicBlock &MBB, unsigned Reg);
-  void populateThunk(MachineFunction &MF, unsigned Reg);
+  RetpolineThunkInserter TI1;
+  LVIThunkInserter TI2;
+  void initTIs(Module &M) {
+    TI1.init(M);
+    TI2.init(M);
+  }
+  bool runTIs(MachineModuleInfo &MMI, MachineFunction &MF) {
+    bool Modified = false;
+    Modified |= TI1.run(MMI, MF);
+    Modified |= TI2.run(MMI, MF);
+    return Modified;
+  }
```

Comment:
I wasn't sure how to do [this](https://github.com/llvm/llvm-project/blob/5b519cf1fc6737054cf90b53667e7ddd3a51225f/llvm/lib/Target/X86/X86IndirectThunks.cpp#L138-L154) in C++11, so I just unrolled it manually.